### PR TITLE
Allow name attribute for <fieldset>

### LINKED ISSUE #42: formCollection View Helper doesn't render name
This issue has been moved from the `zendframework` repository as part of the bug migration program as outlined here - http://framework.zend.com/blog/2016-04-11-issue-closures.html

---

Original Issue: https://api.github.com/repos/zendframework/zendframework/issues/7687
User: @gregGit
Created On: 2016-03-14T16:20:39Z
Updated At: 2016-04-13T07:48:07Z
Body
When using form view helper, the formCollection view helper is used to render fieldset. 
But form some reason that i don't understand this view helper doesn't set the name attribute in the fieldset (unset($attributes['name']); in the render Method).
I belive name is a valid html attributes for fieldset i don't understand why ZF2 never render this tag in a fieldset, and it can be usefull to have it in the DOM.

For test create an action like this :

```
  $fs = new \Zend\Form\Fieldset('my-fieldset');
        $fs->setAttribute('name', 'my-fieldset');
        $fs->add(array(
            'type' => 'text',
            'name' => 'text',
            'options' => array(
                'label' => 'The Text'
            )
        ));

        $fs->add(array(
            'type' => 'text',
            'name' => 'title',
            'options' => array(
                'label' => 'Blog Title'
            )
        ));


        $form=new \Zend\Form\Form('my-form');
        $form->add($fs);
        return new ViewModel(array('form' => $form));
```

and just " echo $this->form($form)" in the view. 

No attribute name for the fieldset is rendered.

---
### Comment

User: @froschdesign
Created On: 2016-04-13T07:48:07Z
Updated At: 2016-04-13T07:48:07Z
Body

> I belive name is a valid html attributes for fieldset

It's only allowed in HTML5.

---

---

Originally posted by @GeeH at https://github.com/zendframework/zend-form/issues/86

### PR DESCRIPTION
**Pull request by Slamdunk**
Close #42

### CODE REVIEW

File: src/View/Helper/FormCollection.php [hunk 1]
```diff
@@ -118,8 +128,10 @@ public function render(ElementInterface $element): string
         // Every collection is wrapped by a fieldset if needed
         if ($this->shouldWrap) {
             $attributes = $element->getAttributes();
-            unset($attributes['name']);
-            $attributesString = $attributes ? ' ' . $this->createAttributesString($attributes) : '';
+            if ($this->getDoctype() === Doctype::XHTML1_STRICT) {
```
Comment (froschdesign) [inline src/View/Helper/FormCollection.php]: In HTML4 there is also no attribute `name` for the `fieldset` element.

Comment (Slamdunk) [reply src/View/Helper/FormCollection.php]: Indeed, I looked up HTML4 strict specs and you're right.
Are you able to tell me which of the supported Doctypes don't have it?

https://github.com/laminas/laminas-view/blob/c34b15f42cac410f35010163de169ab71c39bf73/src/Helper/Doctype.php#L22-L35

```php
class Doctype extends AbstractHelper
{
    /**#@+
     * DocType constants
     */
    const XHTML11             = 'XHTML11';
    const XHTML1_STRICT       = 'XHTML1_STRICT';
    const XHTML1_TRANSITIONAL = 'XHTML1_TRANSITIONAL';
    const XHTML1_FRAMESET     = 'XHTML1_FRAMESET';
    const XHTML1_RDFA         = 'XHTML1_RDFA';
    const XHTML1_RDFA11       = 'XHTML1_RDFA11';
    const XHTML_BASIC1        = 'XHTML_BASIC1';
    const XHTML5              = 'XHTML5';
    const HTML4_STRICT        = 'HTML4_STRICT';
    const HTML4_LOOSE         = 'HTML4_LOOSE';
    const HTML4_FRAMESET      = 'HTML4_FRAMESET';
    const HTML5               = 'HTML5';
    const CUSTOM_XHTML        = 'CUSTOM_XHTML';
    const CUSTOM              = 'CUSTOM';
}
```

Comment (froschdesign) [reply src/View/Helper/FormCollection.php]: The `name` attribute is only allowed for HTML5 and XHTML5. The attribute was introduced with HTML5.

File: test/View/Helper/FormCollectionTest.php [hunk 2]
```diff
@@ -419,4 +420,45 @@ public function testForElementHelperNotInstanceOfHelperInterface(): void
...
+            [Doctype::XHTML1_RDFA,         false],
+            [Doctype::XHTML1_RDFA11,       false],
+            [Doctype::XHTML_BASIC1,        false],
+            [Doctype::XHTML5,              true],
+            [Doctype::HTML4_STRICT,        false],
+            [Doctype::HTML4_LOOSE,         false],
+            [Doctype::HTML4_FRAMESET,      false],
+            [Doctype::HTML5,               true],
+            [Doctype::CUSTOM_XHTML,        false],
+            [Doctype::CUSTOM,              false],
```
Comment (Slamdunk) [inline test/View/Helper/FormCollectionTest.php:461]: @froschdesign is this `dataProvider` correct?

**Review by froschdesign — APPROVED** [review]
🚢  it!

**Pull request merged**